src/test/scala/GoTests.scala

```diff
@@ -1 +1,102 @@
+          case Stone(Black) => "x"
+          case Stone(White) => "o"
+        }
+      }.mkString(" ")
+    }.mkString("|")
+
+  test("5x5 capture") {
+    val positions = Go.playAll(Seq(Point5(1, 2), Point5(1, 1), Point5(2, 1)).map(Move.apply))
+    assert(show(positions.right.get.head) ==
+        """. x . . .
+          |x . . . .
+          |. . . . .
+          |. . . . .
+          |. . . . .""".split('\n').map(_.trim).mkString)
+  }
+
+  test("5x5 3-1 score") {
+    val positions = Go.playAll(Seq(Point5(1, 2), Point5(3, 3), Point5(2, 1)).map(Move.apply))
+    val p = positions.right.get.head
+    assert(show(p) ==
+      """. x . . .
+        |x . . . .
+        |. . o . .
+        |. . . . .
+        |. . . . .""".split('\n').map(_.trim).mkString)
+    assert(p.score(Black) -> p.score(White) == 3 -> 1)
+  }
+
+  test("5x5 9-11 score") {
+    val p = Go.playAll(Seq(
+      Point5(3, 4), Point5(3, 3), Point5(4, 3), Point5(4, 4), Point5(4, 5), Point5(2, 4),
+      Point5(5, 4), Point5(3, 2), Point5(2, 5), Point5(1, 4), Point5(4, 2), Point5(3, 1)
+    ).map(Move.apply)).right.get.head
+    assert(show(p) ==
+      """. . o . .
+        |. . o x .
+        |. . o x .
+        |o o x . x
+        |. x . x .""".split('\n').map(_.trim).mkString)
+    assert(p.score(Black) -> p.score(White) == 9 -> 11)
+  }
+
+  // Optimal play for opening at (3, 4)
+  // http://erikvanderwerf.tengen.nl/5x5/5x5solved.html
+  val movesOptimal34: Seq[Point5] = Seq(
+    (3, 4), (3, 3), (4, 3), (4, 4), (4, 5),
+    (2, 4), (5, 4), (3, 2), (2, 5), (1, 4),
+    (4, 2), (3, 1), (1, 2), (2, 2), (3, 5),
+    (1, 3), (4, 1), (1, 1), (1, 5)
+  ).map(Function.tupled(Point5.apply))
+
+  test("5x5 full game positions") {
+    val expectedPositions: Seq[String] = Seq(
+      ". . . . .|. . . . .|. . . . .|. . x . .|. . . . .",
+      ". . . . .|. . . . .|. . o . .|. . x . .|. . . . .",
+      ". . . . .|. . . . .|. . o x .|. . x . .|. . . . .",
+      ". . . . .|. . . . .|. . o x .|. . x o .|. . . . .",
+      ". . . . .|. . . . .|. . o x .|. . x o .|. . . x .",
+      ". . . . .|. . . . .|. . o x .|. o x o .|. . . x .",
+      ". . . . .|. . . . .|. . o x .|. o x . x|. . . x .",
+      ". . . . .|. . o . .|. . o x .|. o x . x|. . . x .",
+      ". . . . .|. . o . .|. . o x .|. o x . x|. x . x .",
+      ". . . . .|. . o . .|. . o x .|o o x . x|. x . x .",
+      ". . . . .|. . o x .|. . o x .|o o x . x|. x . x .",
+      ". . o . .|. . o x .|. . o x .|o o x . x|. x . x .",
+      ". . o . .|x . o x .|. . o x .|o o x . x|. x . x .",
+      ". . o . .|x o o x .|. . o x .|o o x . x|. x . x .",
+      ". . o . .|x o o x .|. . o x .|o o x . x|. x x x .",
+      ". . o . .|x o o x .|o . o x .|o o x . x|. x x x .",
+      ". . o x .|x o o x .|o . o x .|o o x . x|. x x x .",
+      "o . o x .|. o o x .|o . o x .|o o x . x|. x x x .",
+      "o . o x .|. o o x .|o . o x .|o o x . x|x x x x ."
+    )
+    Go.playAll(movesOptimal34.map(Move.apply)).right.get.reverse
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Usage of get on optional type.](https://www.codacy.com/app/olivier-blanvillain/19x19-net/file/2586331318/issues/source?bid=3301982&fileBranchId=3302581#l86)